lab_4_fill_words_by_ngrams/main.py

```diff
@@ -98,6 +143,42 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
                 or if sequence has inappropriate length,
                 or if methods used return None.
         """
+        if not (isinstance(prompt, str) and prompt
+                and isinstance(seq_len, int) and seq_len > 0
+        ):
+            raise ValueError
+
+        encoded = self._word_processor.encode(prompt)
+
+        if not encoded:
+            raise ValueError
+
+        while seq_len >= 1:
+            tokens = self._model.generate_next_token(encoded)
+            if tokens is None:
+                raise ValueError
+
+            if not tokens:
+                break
+
+            sorted_tokens = sorted(list(tokens.items()), key=lambda pair: (float(pair[1]), -pair[0]), reverse=True)
+            probability = 0
+            for index, token_prob in enumerate(sorted_tokens):
```

Comment:
consider iterating with `index, (token, token_prob)`, this should be more readable